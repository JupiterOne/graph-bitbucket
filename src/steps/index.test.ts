import {
  executeStepWithDependencies,
  Recording,
} from '@jupiterone/integration-sdk-testing';
import { setupBitbucketRecording } from '../../test/recording';
import { buildStepTestConfigForStep } from '../../test/config';

let recording: Recording;
afterEach(async () => {
  await recording.stop();
});

test('fetch-users', async () => {
  recording = setupBitbucketRecording({
    directory: __dirname,
    name: 'fetch-users', //redaction of headers is in setupBitbucketRecording
  });

  const stepConfig = buildStepTestConfigForStep('fetch-users');
  const stepResult = await executeStepWithDependencies(stepConfig);
  expect(stepResult).toMatchStepMetadata(stepConfig);
}, 500000);

test('fetch-workspaces', async () => {
  recording = setupBitbucketRecording({
    directory: __dirname,
    name: 'fetch-workspaces', //redaction of headers is in setupBitbucketRecording
  });

  const stepConfig = buildStepTestConfigForStep('fetch-workspaces');
  const stepResult = await executeStepWithDependencies(stepConfig);
  expect(stepResult).toMatchStepMetadata(stepConfig);
}, 500000);

test('fetch-projects', async () => {
  recording = setupBitbucketRecording({
    directory: __dirname,
    name: 'fetch-projects', //redaction of headers is in setupBitbucketRecording
  });

  const stepConfig = buildStepTestConfigForStep('fetch-projects');
  const stepResult = await executeStepWithDependencies(stepConfig);
  expect(stepResult).toMatchStepMetadata(stepConfig);
}, 500000);

test('fetch-repos', async () => {
  recording = setupBitbucketRecording({
    directory: __dirname,
    name: 'fetch-repos', //redaction of headers is in setupBitbucketRecording
  });

  const stepConfig = buildStepTestConfigForStep('fetch-repos');
  const stepResult = await executeStepWithDependencies(stepConfig);
  expect(stepResult).toMatchStepMetadata(stepConfig);
}, 500000);

test('fetch-repo-permissions', async () => {
  recording = setupBitbucketRecording({
    directory: __dirname,
    name: 'fetch-repo-permissions', //redaction of headers is in setupBitbucketRecording
  });

  const stepConfig = buildStepTestConfigForStep('fetch-repo-permissions');
  const stepResult = await executeStepWithDependencies(stepConfig);
  expect(stepResult).toMatchStepMetadata(stepConfig);
}, 500000);

test('fetch-repo-branch-restrictions', async () => {
  recording = setupBitbucketRecording({
    directory: __dirname,
    name: 'fetch-repo-branch-restrictions', //redaction of headers is in setupBitbucketRecording
  });

  const stepConfig = buildStepTestConfigForStep(
    'fetch-repo-branch-restrictions',
  );
  const stepResult = await executeStepWithDependencies(stepConfig);
  expect(stepResult).toMatchStepMetadata(stepConfig);
}, 500000);
