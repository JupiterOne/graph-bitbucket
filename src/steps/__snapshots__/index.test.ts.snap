// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Account",
      ],
      "_key": "{5ade5777-444d-4a40-bc89-1d43b04487a7}",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "created_on": "2021-03-19T00:44:15.446518+00:00",
            "is_private": true,
            "links": Object {
              "avatar": Object {
                "href": "https://bitbucket.org/workspaces/jupiterone-dev/avatar/?ts=1616114655",
              },
              "hooks": Object {
                "href": "https://bitbucket.org/!api/2.0/workspaces/jupiterone-dev/hooks",
              },
              "html": Object {
                "href": "https://bitbucket.org/jupiterone-dev/",
              },
              "members": Object {
                "href": "https://bitbucket.org/!api/2.0/workspaces/jupiterone-dev/members",
              },
              "owners": Object {
                "href": "https://bitbucket.org/!api/2.0/workspaces/jupiterone-dev/members?q=permission%3D%22owner%22",
              },
              "projects": Object {
                "href": "https://bitbucket.org/!api/2.0/workspaces/jupiterone-dev/projects",
              },
              "repositories": Object {
                "href": "https://bitbucket.org/!api/2.0/repositories/jupiterone-dev",
              },
              "self": Object {
                "href": "https://bitbucket.org/!api/2.0/workspaces/jupiterone-dev",
              },
              "snippets": Object {
                "href": "https://bitbucket.org/!api/2.0/snippets/jupiterone-dev",
              },
            },
            "name": "jupiterone-dev",
            "slug": "jupiterone-dev",
            "type": "workspace",
            "uuid": "{5ade5777-444d-4a40-bc89-1d43b04487a7}",
          },
        },
      ],
      "_type": "bitbucket_workspace",
      "createdOn": 1616114655446,
      "displayName": "jupiterone-dev",
      "isPrivate": true,
      "name": "jupiterone-dev",
      "slug": "jupiterone-dev",
      "updatedOn": undefined,
      "webLink": "https://bitbucket.org/jupiterone-dev/",
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "{89df9e73-a58a-4a26-8db9-8cb1b9d62955}",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "account_id": "5fa964870b4e3a006bfada9f",
            "display_name": "User 0",
            "links": Object {
              "avatar": Object {
                "href": "https://avatar-management--avatars.us-west-2.prod.public.atl-paas.net/5fa964870b4e3a006bfada9f/f10d90e5-290e-4d04-b760-1785a165f6bb/128",
              },
              "html": Object {
                "href": "https://bitbucket.org/%7B89df9e73-a58a-4a26-8db9-8cb1b9d62955%7D/",
              },
              "self": Object {
                "href": "https://bitbucket.org/!api/2.0/users/%7B89df9e73-a58a-4a26-8db9-8cb1b9d62955%7D",
              },
            },
            "nickname": "Nickname 0",
            "type": "user",
            "uuid": "{89df9e73-a58a-4a26-8db9-8cb1b9d62955}",
          },
        },
      ],
      "_type": "bitbucket_user",
      "createdOn": undefined,
      "displayName": "User 0",
      "name": "User 0",
      "nickname": "Nickname 0",
      "username": "User 0",
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "{f1eb42cf-0d64-4bb9-bac5-459b83d9f0ed}",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "account_id": "5f29c97223045b001a70db08",
            "display_name": "User 1",
            "links": Object {
              "avatar": Object {
                "href": "https://avatar-management--avatars.us-west-2.prod.public.atl-paas.net/5f29c97223045b001a70db08/97cf8100-786d-4ce6-b3d0-650961415ec1/128",
              },
              "html": Object {
                "href": "https://bitbucket.org/%7Bf1eb42cf-0d64-4bb9-bac5-459b83d9f0ed%7D/",
              },
              "self": Object {
                "href": "https://bitbucket.org/!api/2.0/users/%7Bf1eb42cf-0d64-4bb9-bac5-459b83d9f0ed%7D",
              },
            },
            "nickname": "Nickname 1",
            "type": "user",
            "uuid": "{f1eb42cf-0d64-4bb9-bac5-459b83d9f0ed}",
          },
        },
      ],
      "_type": "bitbucket_user",
      "createdOn": undefined,
      "displayName": "User 1",
      "name": "User 1",
      "nickname": "Nickname 1",
      "username": "User 1",
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "{2f0561d5-c4c4-4cde-8082-f14a2122cb70}",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "account_id": "5e779055ea2da70c2d23614b",
            "display_name": "User 2",
            "links": Object {
              "avatar": Object {
                "href": "https://avatar-management--avatars.us-west-2.prod.public.atl-paas.net/5e779055ea2da70c2d23614b/e78f3054-e58a-4054-826c-c68470f92e7d/128",
              },
              "html": Object {
                "href": "https://bitbucket.org/%7B2f0561d5-c4c4-4cde-8082-f14a2122cb70%7D/",
              },
              "self": Object {
                "href": "https://bitbucket.org/!api/2.0/users/%7B2f0561d5-c4c4-4cde-8082-f14a2122cb70%7D",
              },
            },
            "nickname": "Nickname 2",
            "type": "user",
            "uuid": "{2f0561d5-c4c4-4cde-8082-f14a2122cb70}",
          },
        },
      ],
      "_type": "bitbucket_user",
      "createdOn": undefined,
      "displayName": "User 2",
      "name": "User 2",
      "nickname": "Nickname 2",
      "username": "User 2",
    },
    Object {
      "_class": Array [
        "UserGroup",
      ],
      "_key": "bitbucket-group:developers",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "auto_add": true,
            "email_forwarding_disabled": null,
            "name": "Developers",
            "owner": Object {
              "account_id": null,
              "avatar": "https://bitbucket.org/account/jupiterone-dev/avatar/?ts=1616114658",
              "display_name": "jupiterone-dev",
              "is_team": true,
              "nickname": "jupiterone-dev",
              "resource_uri": "/!api/1.0/users/%7B5ade5777-444d-4a40-bc89-1d43b04487a7%7D",
              "uuid": "{5ade5777-444d-4a40-bc89-1d43b04487a7}",
            },
            "permission": "write",
            "slug": "developers",
          },
        },
      ],
      "_type": "bitbucket_group",
      "autoAdd": true,
      "createdOn": undefined,
      "displayName": "Developers",
      "name": "Developers",
      "permission": "write",
      "slug": "developers",
    },
    Object {
      "_class": Array [
        "UserGroup",
      ],
      "_key": "bitbucket-group:administrators",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "auto_add": true,
            "email_forwarding_disabled": null,
            "name": "Administrators",
            "owner": Object {
              "account_id": null,
              "avatar": "https://bitbucket.org/account/jupiterone-dev/avatar/?ts=1616114658",
              "display_name": "jupiterone-dev",
              "is_team": true,
              "nickname": "jupiterone-dev",
              "resource_uri": "/!api/1.0/users/%7B5ade5777-444d-4a40-bc89-1d43b04487a7%7D",
              "uuid": "{5ade5777-444d-4a40-bc89-1d43b04487a7}",
            },
            "permission": "admin",
            "slug": "administrators",
          },
        },
      ],
      "_type": "bitbucket_group",
      "autoAdd": true,
      "createdOn": undefined,
      "displayName": "Administrators",
      "name": "Administrators",
      "permission": "admin",
      "slug": "administrators",
    },
    Object {
      "_class": Array [
        "Project",
      ],
      "_key": "{e608ae89-60af-45dc-bab1-66bccc8a295a}",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "created_on": "2021-03-19T00:46:29.368383+00:00",
            "description": null,
            "is_private": false,
            "key": "PROJ-2",
            "links": Object {
              "avatar": Object {
                "href": "https://bitbucket.org/account/user/jupiterone-dev/projects/PROJ-2/avatar/32?ts=1616114789",
              },
              "html": Object {
                "href": "https://bitbucket.org/jupiterone-dev/workspace/projects/PROJ-2",
              },
              "repositories": Object {
                "href": "https://bitbucket.org/!api/2.0/repositories/jupiterone-dev?q=project.key=\\"PROJ-2\\"",
              },
              "self": Object {
                "href": "https://bitbucket.org/!api/2.0/workspaces/jupiterone-dev/projects/PROJ-2",
              },
            },
            "name": "ProjectB",
            "owner": Object {
              "display_name": "jupiterone-dev",
              "links": Object {
                "avatar": Object {
                  "href": "https://bitbucket.org/account/jupiterone-dev/avatar/",
                },
                "html": Object {
                  "href": "https://bitbucket.org/%7B5ade5777-444d-4a40-bc89-1d43b04487a7%7D/",
                },
                "self": Object {
                  "href": "https://bitbucket.org/!api/2.0/teams/%7B5ade5777-444d-4a40-bc89-1d43b04487a7%7D",
                },
              },
              "type": "team",
              "username": "jupiterone-dev",
              "uuid": "{5ade5777-444d-4a40-bc89-1d43b04487a7}",
            },
            "type": "project",
            "updated_on": "2021-03-19T00:46:29.368394+00:00",
            "uuid": "{e608ae89-60af-45dc-bab1-66bccc8a295a}",
            "workspace": Object {
              "links": Object {
                "avatar": Object {
                  "href": "https://bitbucket.org/workspaces/jupiterone-dev/avatar/?ts=1616114655",
                },
                "html": Object {
                  "href": "https://bitbucket.org/jupiterone-dev/",
                },
                "self": Object {
                  "href": "https://bitbucket.org/!api/2.0/workspaces/jupiterone-dev",
                },
              },
              "name": "jupiterone-dev",
              "slug": "jupiterone-dev",
              "type": "workspace",
              "uuid": "{5ade5777-444d-4a40-bc89-1d43b04487a7}",
            },
          },
        },
      ],
      "_type": "bitbucket_project",
      "createdOn": 1616114789368,
      "description": "",
      "displayName": "ProjectB",
      "key": "PROJ-2",
      "name": "ProjectB",
      "public": true,
      "updatedOn": 1616114789368,
      "webLink": "https://bitbucket.org/jupiterone-dev/workspace/projects/PROJ-2",
      "workspace": "jupiterone-dev",
    },
    Object {
      "_class": Array [
        "Project",
      ],
      "_key": "{819a7369-1044-444e-921c-0124fec32830}",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "created_on": "2021-03-19T00:45:45.638134+00:00",
            "description": null,
            "is_private": false,
            "key": "PROJ",
            "links": Object {
              "avatar": Object {
                "href": "https://bitbucket.org/account/user/jupiterone-dev/projects/PROJ/avatar/32?ts=1616114745",
              },
              "html": Object {
                "href": "https://bitbucket.org/jupiterone-dev/workspace/projects/PROJ",
              },
              "repositories": Object {
                "href": "https://bitbucket.org/!api/2.0/repositories/jupiterone-dev?q=project.key=\\"PROJ\\"",
              },
              "self": Object {
                "href": "https://bitbucket.org/!api/2.0/workspaces/jupiterone-dev/projects/PROJ",
              },
            },
            "name": "ProjectA",
            "owner": Object {
              "display_name": "jupiterone-dev",
              "links": Object {
                "avatar": Object {
                  "href": "https://bitbucket.org/account/jupiterone-dev/avatar/",
                },
                "html": Object {
                  "href": "https://bitbucket.org/%7B5ade5777-444d-4a40-bc89-1d43b04487a7%7D/",
                },
                "self": Object {
                  "href": "https://bitbucket.org/!api/2.0/teams/%7B5ade5777-444d-4a40-bc89-1d43b04487a7%7D",
                },
              },
              "type": "team",
              "username": "jupiterone-dev",
              "uuid": "{5ade5777-444d-4a40-bc89-1d43b04487a7}",
            },
            "type": "project",
            "updated_on": "2021-03-19T00:45:45.638155+00:00",
            "uuid": "{819a7369-1044-444e-921c-0124fec32830}",
            "workspace": Object {
              "links": Object {
                "avatar": Object {
                  "href": "https://bitbucket.org/workspaces/jupiterone-dev/avatar/?ts=1616114655",
                },
                "html": Object {
                  "href": "https://bitbucket.org/jupiterone-dev/",
                },
                "self": Object {
                  "href": "https://bitbucket.org/!api/2.0/workspaces/jupiterone-dev",
                },
              },
              "name": "jupiterone-dev",
              "slug": "jupiterone-dev",
              "type": "workspace",
              "uuid": "{5ade5777-444d-4a40-bc89-1d43b04487a7}",
            },
          },
        },
      ],
      "_type": "bitbucket_project",
      "createdOn": 1616114745638,
      "description": "",
      "displayName": "ProjectA",
      "key": "PROJ",
      "name": "ProjectA",
      "public": true,
      "updatedOn": 1616114745638,
      "webLink": "https://bitbucket.org/jupiterone-dev/workspace/projects/PROJ",
      "workspace": "jupiterone-dev",
    },
    Object {
      "_class": Array [
        "CodeRepo",
      ],
      "_key": "{2408f9ea-6d10-4897-84c5-aeeffb1b4c14}",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "created_on": "2021-03-19T00:45:45.659908+00:00",
            "description": "",
            "fork_policy": "no_public_forks",
            "full_name": "jupiterone-dev/project-a-repo-a",
            "has_issues": false,
            "has_wiki": false,
            "is_private": true,
            "language": "",
            "links": Object {
              "avatar": Object {
                "href": "https://bytebucket.org/ravatar/%7B2408f9ea-6d10-4897-84c5-aeeffb1b4c14%7D?ts=default",
              },
              "branches": Object {
                "href": "https://bitbucket.org/!api/2.0/repositories/jupiterone-dev/project-a-repo-a/refs/branches",
              },
              "clone": Array [
                Object {
                  "href": "https://jupiterone-dev@bitbucket.org/jupiterone-dev/project-a-repo-a.git",
                  "name": "https",
                },
                Object {
                  "href": "git@bitbucket.org:jupiterone-dev/project-a-repo-a.git",
                  "name": "ssh",
                },
              ],
              "commits": Object {
                "href": "https://bitbucket.org/!api/2.0/repositories/jupiterone-dev/project-a-repo-a/commits",
              },
              "downloads": Object {
                "href": "https://bitbucket.org/!api/2.0/repositories/jupiterone-dev/project-a-repo-a/downloads",
              },
              "forks": Object {
                "href": "https://bitbucket.org/!api/2.0/repositories/jupiterone-dev/project-a-repo-a/forks",
              },
              "hooks": Object {
                "href": "https://bitbucket.org/!api/2.0/repositories/jupiterone-dev/project-a-repo-a/hooks",
              },
              "html": Object {
                "href": "https://bitbucket.org/jupiterone-dev/project-a-repo-a",
              },
              "pullrequests": Object {
                "href": "https://bitbucket.org/!api/2.0/repositories/jupiterone-dev/project-a-repo-a/pullrequests",
              },
              "self": Object {
                "href": "https://bitbucket.org/!api/2.0/repositories/jupiterone-dev/project-a-repo-a",
              },
              "source": Object {
                "href": "https://bitbucket.org/!api/2.0/repositories/jupiterone-dev/project-a-repo-a/src",
              },
              "tags": Object {
                "href": "https://bitbucket.org/!api/2.0/repositories/jupiterone-dev/project-a-repo-a/refs/tags",
              },
              "watchers": Object {
                "href": "https://bitbucket.org/!api/2.0/repositories/jupiterone-dev/project-a-repo-a/watchers",
              },
            },
            "mainbranch": Object {
              "name": "master",
              "type": "branch",
            },
            "name": "project-a-repo-a",
            "owner": Object {
              "display_name": "jupiterone-dev",
              "links": Object {
                "avatar": Object {
                  "href": "https://bitbucket.org/account/jupiterone-dev/avatar/",
                },
                "html": Object {
                  "href": "https://bitbucket.org/%7B5ade5777-444d-4a40-bc89-1d43b04487a7%7D/",
                },
                "self": Object {
                  "href": "https://bitbucket.org/!api/2.0/teams/%7B5ade5777-444d-4a40-bc89-1d43b04487a7%7D",
                },
              },
              "type": "team",
              "username": "jupiterone-dev",
              "uuid": "{5ade5777-444d-4a40-bc89-1d43b04487a7}",
            },
            "project": Object {
              "key": "PROJ",
              "links": Object {
                "avatar": Object {
                  "href": "https://bitbucket.org/account/user/jupiterone-dev/projects/PROJ/avatar/32?ts=1616114745",
                },
                "html": Object {
                  "href": "https://bitbucket.org/jupiterone-dev/workspace/projects/PROJ",
                },
                "self": Object {
                  "href": "https://bitbucket.org/!api/2.0/workspaces/jupiterone-dev/projects/PROJ",
                },
              },
              "name": "ProjectA",
              "type": "project",
              "uuid": "{819a7369-1044-444e-921c-0124fec32830}",
            },
            "scm": "git",
            "size": 89247,
            "slug": "project-a-repo-a",
            "type": "repository",
            "updated_on": "2021-05-19T18:49:34.699444+00:00",
            "uuid": "{2408f9ea-6d10-4897-84c5-aeeffb1b4c14}",
            "website": null,
            "workspace": Object {
              "links": Object {
                "avatar": Object {
                  "href": "https://bitbucket.org/workspaces/jupiterone-dev/avatar/?ts=1616114655",
                },
                "html": Object {
                  "href": "https://bitbucket.org/jupiterone-dev/",
                },
                "self": Object {
                  "href": "https://bitbucket.org/!api/2.0/workspaces/jupiterone-dev",
                },
              },
              "name": "jupiterone-dev",
              "slug": "jupiterone-dev",
              "type": "workspace",
              "uuid": "{5ade5777-444d-4a40-bc89-1d43b04487a7}",
            },
          },
        },
      ],
      "_type": "bitbucket_repo",
      "createdOn": 1616114745659,
      "description": "",
      "displayName": "project-a-repo-a",
      "fullName": "jupiterone-dev/project-a-repo-a",
      "name": "project-a-repo-a",
      "owner": "jupiterone-dev",
      "ownerId": "{5ade5777-444d-4a40-bc89-1d43b04487a7}",
      "projectId": "{819a7369-1044-444e-921c-0124fec32830}",
      "public": false,
      "updatedOn": 1621450174699,
      "webLink": "https://bitbucket.org/jupiterone-dev/project-a-repo-a",
    },
    Object {
      "_class": Array [
        "CodeRepo",
      ],
      "_key": "{d052842f-c89c-4690-9e73-5162e0f23be1}",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "created_on": "2021-03-19T00:46:29.387599+00:00",
            "description": "",
            "fork_policy": "no_public_forks",
            "full_name": "jupiterone-dev/project-b-repo-a",
            "has_issues": false,
            "has_wiki": false,
            "is_private": true,
            "language": "",
            "links": Object {
              "avatar": Object {
                "href": "https://bytebucket.org/ravatar/%7Bd052842f-c89c-4690-9e73-5162e0f23be1%7D?ts=default",
              },
              "branches": Object {
                "href": "https://bitbucket.org/!api/2.0/repositories/jupiterone-dev/project-b-repo-a/refs/branches",
              },
              "clone": Array [
                Object {
                  "href": "https://jupiterone-dev@bitbucket.org/jupiterone-dev/project-b-repo-a.git",
                  "name": "https",
                },
                Object {
                  "href": "git@bitbucket.org:jupiterone-dev/project-b-repo-a.git",
                  "name": "ssh",
                },
              ],
              "commits": Object {
                "href": "https://bitbucket.org/!api/2.0/repositories/jupiterone-dev/project-b-repo-a/commits",
              },
              "downloads": Object {
                "href": "https://bitbucket.org/!api/2.0/repositories/jupiterone-dev/project-b-repo-a/downloads",
              },
              "forks": Object {
                "href": "https://bitbucket.org/!api/2.0/repositories/jupiterone-dev/project-b-repo-a/forks",
              },
              "hooks": Object {
                "href": "https://bitbucket.org/!api/2.0/repositories/jupiterone-dev/project-b-repo-a/hooks",
              },
              "html": Object {
                "href": "https://bitbucket.org/jupiterone-dev/project-b-repo-a",
              },
              "pullrequests": Object {
                "href": "https://bitbucket.org/!api/2.0/repositories/jupiterone-dev/project-b-repo-a/pullrequests",
              },
              "self": Object {
                "href": "https://bitbucket.org/!api/2.0/repositories/jupiterone-dev/project-b-repo-a",
              },
              "source": Object {
                "href": "https://bitbucket.org/!api/2.0/repositories/jupiterone-dev/project-b-repo-a/src",
              },
              "tags": Object {
                "href": "https://bitbucket.org/!api/2.0/repositories/jupiterone-dev/project-b-repo-a/refs/tags",
              },
              "watchers": Object {
                "href": "https://bitbucket.org/!api/2.0/repositories/jupiterone-dev/project-b-repo-a/watchers",
              },
            },
            "mainbranch": Object {
              "name": "master",
              "type": "branch",
            },
            "name": "project-b-repo-a",
            "owner": Object {
              "display_name": "jupiterone-dev",
              "links": Object {
                "avatar": Object {
                  "href": "https://bitbucket.org/account/jupiterone-dev/avatar/",
                },
                "html": Object {
                  "href": "https://bitbucket.org/%7B5ade5777-444d-4a40-bc89-1d43b04487a7%7D/",
                },
                "self": Object {
                  "href": "https://bitbucket.org/!api/2.0/teams/%7B5ade5777-444d-4a40-bc89-1d43b04487a7%7D",
                },
              },
              "type": "team",
              "username": "jupiterone-dev",
              "uuid": "{5ade5777-444d-4a40-bc89-1d43b04487a7}",
            },
            "project": Object {
              "key": "PROJ-2",
              "links": Object {
                "avatar": Object {
                  "href": "https://bitbucket.org/account/user/jupiterone-dev/projects/PROJ-2/avatar/32?ts=1616114789",
                },
                "html": Object {
                  "href": "https://bitbucket.org/jupiterone-dev/workspace/projects/PROJ-2",
                },
                "self": Object {
                  "href": "https://bitbucket.org/!api/2.0/workspaces/jupiterone-dev/projects/PROJ-2",
                },
              },
              "name": "ProjectB",
              "type": "project",
              "uuid": "{e608ae89-60af-45dc-bab1-66bccc8a295a}",
            },
            "scm": "git",
            "size": 63370,
            "slug": "project-b-repo-a",
            "type": "repository",
            "updated_on": "2021-03-19T00:46:30.031651+00:00",
            "uuid": "{d052842f-c89c-4690-9e73-5162e0f23be1}",
            "website": null,
            "workspace": Object {
              "links": Object {
                "avatar": Object {
                  "href": "https://bitbucket.org/workspaces/jupiterone-dev/avatar/?ts=1616114655",
                },
                "html": Object {
                  "href": "https://bitbucket.org/jupiterone-dev/",
                },
                "self": Object {
                  "href": "https://bitbucket.org/!api/2.0/workspaces/jupiterone-dev",
                },
              },
              "name": "jupiterone-dev",
              "slug": "jupiterone-dev",
              "type": "workspace",
              "uuid": "{5ade5777-444d-4a40-bc89-1d43b04487a7}",
            },
          },
        },
      ],
      "_type": "bitbucket_repo",
      "createdOn": 1616114789387,
      "description": "",
      "displayName": "project-b-repo-a",
      "fullName": "jupiterone-dev/project-b-repo-a",
      "name": "project-b-repo-a",
      "owner": "jupiterone-dev",
      "ownerId": "{5ade5777-444d-4a40-bc89-1d43b04487a7}",
      "projectId": "{e608ae89-60af-45dc-bab1-66bccc8a295a}",
      "public": false,
      "updatedOn": 1616114790031,
      "webLink": "https://bitbucket.org/jupiterone-dev/project-b-repo-a",
    },
  ],
  "collectedRelationships": Array [
    Object {
      "_class": "HAS",
      "_fromEntityKey": "{5ade5777-444d-4a40-bc89-1d43b04487a7}",
      "_key": "{5ade5777-444d-4a40-bc89-1d43b04487a7}|has|{89df9e73-a58a-4a26-8db9-8cb1b9d62955}",
      "_toEntityKey": "{89df9e73-a58a-4a26-8db9-8cb1b9d62955}",
      "_type": "bitbucket_workspace_has_user",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "{5ade5777-444d-4a40-bc89-1d43b04487a7}",
      "_key": "{5ade5777-444d-4a40-bc89-1d43b04487a7}|has|{f1eb42cf-0d64-4bb9-bac5-459b83d9f0ed}",
      "_toEntityKey": "{f1eb42cf-0d64-4bb9-bac5-459b83d9f0ed}",
      "_type": "bitbucket_workspace_has_user",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "{5ade5777-444d-4a40-bc89-1d43b04487a7}",
      "_key": "{5ade5777-444d-4a40-bc89-1d43b04487a7}|has|{2f0561d5-c4c4-4cde-8082-f14a2122cb70}",
      "_toEntityKey": "{2f0561d5-c4c4-4cde-8082-f14a2122cb70}",
      "_type": "bitbucket_workspace_has_user",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "{5ade5777-444d-4a40-bc89-1d43b04487a7}",
      "_key": "{5ade5777-444d-4a40-bc89-1d43b04487a7}|has|bitbucket-group:developers",
      "_toEntityKey": "bitbucket-group:developers",
      "_type": "bitbucket_workspace_has_group",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "{5ade5777-444d-4a40-bc89-1d43b04487a7}",
      "_key": "{5ade5777-444d-4a40-bc89-1d43b04487a7}|has|bitbucket-group:administrators",
      "_toEntityKey": "bitbucket-group:administrators",
      "_type": "bitbucket_workspace_has_group",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "bitbucket-group:administrators",
      "_key": "bitbucket-group:administrators|has|{f1eb42cf-0d64-4bb9-bac5-459b83d9f0ed}",
      "_toEntityKey": "{f1eb42cf-0d64-4bb9-bac5-459b83d9f0ed}",
      "_type": "bitbucket_group_has_user",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "bitbucket-group:administrators",
      "_key": "bitbucket-group:administrators|has|{89df9e73-a58a-4a26-8db9-8cb1b9d62955}",
      "_toEntityKey": "{89df9e73-a58a-4a26-8db9-8cb1b9d62955}",
      "_type": "bitbucket_group_has_user",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "bitbucket-group:administrators",
      "_key": "bitbucket-group:administrators|has|{2f0561d5-c4c4-4cde-8082-f14a2122cb70}",
      "_toEntityKey": "{2f0561d5-c4c4-4cde-8082-f14a2122cb70}",
      "_type": "bitbucket_group_has_user",
      "displayName": "HAS",
    },
    Object {
      "_class": "OWNS",
      "_fromEntityKey": "{5ade5777-444d-4a40-bc89-1d43b04487a7}",
      "_key": "{5ade5777-444d-4a40-bc89-1d43b04487a7}|owns|{e608ae89-60af-45dc-bab1-66bccc8a295a}",
      "_toEntityKey": "{e608ae89-60af-45dc-bab1-66bccc8a295a}",
      "_type": "bitbucket_workspace_owns_project",
      "displayName": "OWNS",
    },
    Object {
      "_class": "OWNS",
      "_fromEntityKey": "{5ade5777-444d-4a40-bc89-1d43b04487a7}",
      "_key": "{5ade5777-444d-4a40-bc89-1d43b04487a7}|owns|{819a7369-1044-444e-921c-0124fec32830}",
      "_toEntityKey": "{819a7369-1044-444e-921c-0124fec32830}",
      "_type": "bitbucket_workspace_owns_project",
      "displayName": "OWNS",
    },
    Object {
      "_class": "OWNS",
      "_fromEntityKey": "{5ade5777-444d-4a40-bc89-1d43b04487a7}",
      "_key": "{5ade5777-444d-4a40-bc89-1d43b04487a7}|owns|{2408f9ea-6d10-4897-84c5-aeeffb1b4c14}",
      "_toEntityKey": "{2408f9ea-6d10-4897-84c5-aeeffb1b4c14}",
      "_type": "bitbucket_workspace_owns_repo",
      "displayName": "OWNS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "{819a7369-1044-444e-921c-0124fec32830}",
      "_key": "{819a7369-1044-444e-921c-0124fec32830}|has|{2408f9ea-6d10-4897-84c5-aeeffb1b4c14}",
      "_toEntityKey": "{2408f9ea-6d10-4897-84c5-aeeffb1b4c14}",
      "_type": "bitbucket_project_has_repo",
      "displayName": "HAS",
    },
    Object {
      "_class": "OWNS",
      "_fromEntityKey": "{5ade5777-444d-4a40-bc89-1d43b04487a7}",
      "_key": "{5ade5777-444d-4a40-bc89-1d43b04487a7}|owns|{d052842f-c89c-4690-9e73-5162e0f23be1}",
      "_toEntityKey": "{d052842f-c89c-4690-9e73-5162e0f23be1}",
      "_type": "bitbucket_workspace_owns_repo",
      "displayName": "OWNS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "{e608ae89-60af-45dc-bab1-66bccc8a295a}",
      "_key": "{e608ae89-60af-45dc-bab1-66bccc8a295a}|has|{d052842f-c89c-4690-9e73-5162e0f23be1}",
      "_toEntityKey": "{d052842f-c89c-4690-9e73-5162e0f23be1}",
      "_type": "bitbucket_project_has_repo",
      "displayName": "HAS",
    },
  ],
  "encounteredTypes": Array [
    "bitbucket_workspace",
    "bitbucket_user",
    "bitbucket_workspace_has_user",
    "bitbucket_group",
    "bitbucket_workspace_has_group",
    "bitbucket_group_has_user",
    "bitbucket_project",
    "bitbucket_workspace_owns_project",
    "bitbucket_repo",
    "bitbucket_workspace_owns_repo",
    "bitbucket_project_has_repo",
  ],
  "numCollectedEntities": 10,
  "numCollectedRelationships": 14,
}
`;
